schema.go

```diff
@@ -592,6 +673,11 @@ func (s *RecordSchema) Type() Type {
 	return Record
 }
 
+// Logical returns the logical schema or nil.
+func (s *RecordSchema) Logical() LogicalSchema {
```

Comment:
This means that every schema now implements `LogicalTypeSchema`. From a spec PoV this is really confusing.